src/main/java/fi/helsinki/cs/tmc/core/domain/Course.java

```diff
@@ -67,6 +71,39 @@ public void setExercises(List<Exercise> exercises) {
         this.exercises = exercises;
     }
 
+    public List<Theme> getThemes() {
+        return themes;
+    }
+
+    public void generateThemes() {
+        themes = new ArrayList<>();
+        Map<String, Theme> themeMap = new HashMap<>();
+        for (Exercise ex : exercises) {
+            addExerciseToTheme(themeMap, ex);
+        }
+    }
+
+    private void addExerciseToTheme(Map<String, Theme> themeMap, Exercise ex) {
+        String themeName = ex.getName().split("-")[0];
+        Theme theme = themeMap.get(themeName);
+        if (theme == null) {
+            theme = new Theme(themeName);
+            themeMap.put(themeName, theme);
+            themes.add(theme);
+        }
+        theme.addExercise(ex);
+    }
+
+    public List<Exercise> getExercisesByTheme(Theme theme) {
+        ArrayList<Exercise> exercisesByTheme = new ArrayList<>();
+        for (Exercise exercise : theme.getExercises()) {
+            if (this.exercises.contains(exercise)) {
+                exercisesByTheme.add(exercise);
+            }
+        }
+        return exercisesByTheme;
+    }
```

Comment:
A theme (such as "For-loop") might have exercises related to multiple courses. As a user I can get an adaptive exercise when I have complited all exercises of that theme, so I wrote this method to help checking it in tmc-netbeans.